src/Core/src/Platform/iOS/ViewExtensions.cs

```diff
@@ -284,14 +284,70 @@ static void UpdateBackgroundLayers(this CALayer[] layers, string layerName, CGRe
 			}
 		}
 
+		/// <summary>
+		/// Invalidates the measure of the view and all its ancestors through <see cref="UIView.SetNeedsLayout"/> propagation.
+		/// </summary>
+		/// <remarks>
+		/// Stops when it reaches the page view or a scrollable area, including <see cref="UICollectionView"/>.
+		/// </remarks>
 		public static void InvalidateMeasure(this UIView platformView, IView view)
 		{
-			platformView.SetNeedsLayout();
+			if (platformView is IMauiPlatformView mauiPlatformView)
+			{
+				mauiPlatformView.InvalidateMeasure();
+			}
+			else
+			{
+				platformView.SetNeedsLayout();
+			}
+
+			platformView.InvalidateAncestorsMeasures();
+		}
 
-			// MauiView/WrapperView already propagates the SetNeedsLayout to the parent
-			if (platformView is not MauiView && platformView is not WrapperView)
+		internal static void InvalidateAncestorsMeasures(this UIView child)
+		{
+			var childMauiPlatformLayout = child as IMauiPlatformView;
+
+			while (true)
 			{
-				platformView.Superview?.SetNeedsLayout();
+				// We verify the presence of a Window to prevent scenarios where an invalidate might propagate up the view hierarchy  
+				// to a SuperView that has already been disposed. Accessing such a disposed view would result in a crash (see #24032).  
+				// This validation is only possible using `IMauiPlatformView`, as it provides a way to schedule an invalidation when the view is moved to window.  
+				// For other cases, we accept the risk since avoiding it could lead to the layout not being updated properly.
+				if (childMauiPlatformLayout != null && child.Window is null)
+				{
+					childMauiPlatformLayout.InvalidateAncestorsMeasuresWhenMovedToWindow();
+					return;
+				}
+
+				var superview = child.Superview;
+				if (superview is null)
+				{
+					return;
+				}
+
+				// Now invalidate the parent view
+				var superviewMauiPlatformLayout = superview as IMauiPlatformView;
+				if (superviewMauiPlatformLayout is not null)
+				{
+					superviewMauiPlatformLayout.InvalidateMeasure(isPropagating: true);
+				}
+				else
+				{
+					superview.SetNeedsLayout();
+				}
+
+				// Potential improvement: if the MAUI view (superview here) is constrained to a fixed size, we could stop propagating
+				// when doing this, we must pay attention to a scenario where a non-fixed-size view becomes fixed-size
+				if (superview is PageView or UIScrollView)
```

Comment:
What happens if the scroll view is in a VSL? Technically it has infinite size. Evil, but it can't explode.

Reply:
@mattleibow the `MauiScrollView` will invalidate itself.
CV1 and CV2 do the same.

```
				// When the content size changes, we need to adjust the scrollable area size so that the content can fit in it.
				if (ContentSize != contentSize)
				{
					ContentSize = contentSize;

					// Invalidation stops at `UIScrollViews` for performance reasons,
					// but when the content size changes, we need to invalidate the ancestors
					// in case the ScrollView is configured to grow/shrink with its content.
					this.InvalidateAncestorsMeasures();
				}
```